### PR TITLE
eza: create exa compatiblity symlink

### PR DESCRIPTION
**Pull request by peterhoeg**
## Description of changes

Further to discussion on https://github.com/NixOS/nixpkgs/pull/253683 post merge.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/eza/default.nix [hunk 1]
```diff
@@ -10,8 +10,12 @@
 , Security
 , libiconv
 , installShellFiles
+, exaAlias ? true
 }:
 
+# once eza upstream gets support for setting up a compatibilty symlink for exa, we should change
```
Comment (Mindavi) [inline pkgs/tools/misc/eza/default.nix:16]: Probably best to put this comment above the option?

**Review by Mindavi — APPROVED** [review]
Makes sense to me.

**Comment by peterhoeg** [comment]
I'll squash the commits when done.

File: pkgs/tools/misc/eza/default.nix [hunk 2]
```diff
@@ -43,6 +46,8 @@ rustPlatform.buildRustPackage rec {
       --bash completions/bash/eza \
       --fish completions/fish/eza.fish \
       --zsh completions/zsh/_eza
+  '' + lib.optionalString exaAlias ''
+    ln -s $out/bin/eza $out/bin/exa
```
Comment (oxalica) [inline pkgs/tools/misc/eza/default.nix:50]: To be simpler:

```suggestion
    ln -s eza $out/bin/exa
```

It'll be rewritten to be relative anyway.

> `eza> rewriting symlink /nix/store/pdy2gxifbz4xp55wvf3y6a03g2bv6kwf-eza-0.11.0/bin/exa to be relative to /nix/store/pdy2gxifbz4xp55wvf3y6a03g2bv6kwf-eza-0.11.0`

**Review by cafkafk — APPROVED** [review]

**Pull request merged**